scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+		if (this.config.paths) {
+			let response: LinkResponse = {
+				linkedPaths: [],
+				alreadyLinkedPaths: [],
+			};
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					response.alreadyLinkedPaths.push(pathName);
+				} else {
+					try {
+						await this.createLink(path);
+						response.linkedPaths.push(pathName);
+					} catch (e) {
+						throw new Error("Could not link path: " + pathName + ": " + e);
+					}
+				}
+			}
+			return response;
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	public async start(paths: string[]): Promise<StartResponse> {
+		if (this.config.paths) {
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					const link = await this.getLink(path);
+					this.relayers.set(
+						pathName,
+						setInterval(async () => {
+							let heights = this.pathById(pathName).relayerData;
+							let newHeights = await this.relayPackets(link, heights);
+							this.pathById(pathName).relayerData = newHeights;
+						}, 5000)
+					);
+				} else {
+					throw new Error("Path: " + pathName + " is not linked.");
+				}
+			}
+			return {};
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	// Helper functions
+	protected chainById(chainID: string): ChainConfig {
+		return this.config.chains
+			? this.config.chains.find((x) => x.chainId == chainID)
+			: null;
+	}
+
+	protected pathById(pathID: string): PathConfig {
+		return this.config.paths
+			? this.config.paths.find((x) => x.path.id == pathID)
+			: null;
+	}
+	protected async balanceCheck(chain: ChainConfig): Promise<boolean> {
+		let chainBalances = await this.getDefaultAccountBalance([chain.chainId]);
+		let chainGP = GasPrice.fromString(chain.gasPrice);
+		if (!chainBalances.find((x) => x.denom == chainGP.denom)) {
+			return false;
+		}
+		if (
+			chainBalances.find(
+				(x) =>
+					x.denom == chainGP.denom &&
+					parseInt(x.amount) < chainGP.amount.toFloatApproximation() * 2256000
+			)
+		) {
+			return false;
+		}
+		return true;
+	}
+	protected async createLink({ path, options }: PathConfig): Promise<void> {
+		let chainA = this.chainById(path.src.chainID);
+		let chainB = this.chainById(path.dst.chainID);
+		let chainAGP = GasPrice.fromString(chainA.gasPrice);
+		let chainBGP = GasPrice.fromString(chainA.gasPrice);
+		if (!(await this.balanceCheck(chainA))) {
+			throw new Error(
+				"Not enough balance available on '" +
+					chainA.chainId +
+					"'. You need at least " +
+					chainAGP.amount.toFloatApproximation() * 2256000 +
+					chainAGP.denom
+			);
+		}
+		if (!(await this.balanceCheck(chainB))) {
+			throw new Error(
+				"Not enough balance available on '" +
+					chainB.chainId +
+					"'. You need at least " +
+					chainBGP.amount.toFloatApproximation() * 2256000 +
+					chainBGP.denom
+			);
+		}
+		// Create IBC Client for chain A
+
+		const clientA = await this.getIBCClient(chainA);
```

Comment:
Let's change A & B to Src & Dst since this is how they're defined in the ibc spec.

Reply:
true but A and B is what is used in a lot of places in the ts-relayer source and since we're using it as a lib it makes it easier for someone to follow the ts-relayer functions we're using after